src/main/scala/analysis/VSA.scala

```diff
@@ -105,239 +68,97 @@ trait MemoryRegionValueSetAnalysis:
-                case _ =>
-                  Logger.error("CASE NOT HANDLED: " + assigmentsMap.get(variable, pred) + " FOR " + binOp)
-                  return exp
-              }
-            }
-          case _ => return evaluateExpression(binOp.arg1, n)
-        }
-        exp
-      case memLoad: MemoryLoad =>
-        evaluateExpression(memLoad.index, n)
-      case bitVecLiteral: BitVecLiteral =>
-        bitVecLiteral
-      case extend: ZeroExtend =>
-        evaluateExpression(extend.body, n)
-      case variable: Variable =>
-        loopEscapeSet.clear()
-        val ssss = findDecl(variable, n)
-        loopEscapeSet.clear()
-        for (pred <- findDecl(variable, n)) {
-          if (!assigmentsMap.contains((variable, pred))) {
-            pred match
-              case cmd: CfgCommandNode =>
-                cmd.data match
-                  case localAssign: LocalAssign =>
-                    assigmentsMap.addOne((localAssign.lhs, pred) -> evaluateExpression(localAssign.rhs, pred))
-                    return evaluateExpression(localAssign.rhs, pred)
-          } else {
-            assigmentsMap(variable, pred) match
-              case bitVecLiteral: BitVecLiteral =>
-                return bitVecLiteral
-              case any: Expr => return exp
-          }
-        }
-        exp
-      case extract: Extract =>
-        evaluateExpression(extract.body, n)
-      case _ =>
-        //throw new RuntimeException("ERROR: CASE NOT HANDLED: " + exp + "\n")
-        exp
-    }
-  }
-
-
   def exprToRegion(expr: Expr, n: CfgNode): Option[MemoryRegion] = {
     expr match
       case binOp: BinaryExpr =>
-        val lhs: Expr = if binOp.arg1.equals(stackPointer) then binOp.arg1 else evaluateExpression(binOp.arg1, n)
-        val rhs: Expr = evaluateExpression(binOp.arg2, n)
-        if (lhs.equals(stackPointer)) {
-          mmm.findObject(rhs.asInstanceOf[BitVecLiteral].value, "stack") match
-            case Some(obj: MemoryRegion) =>
-              return Some(obj)
-            case _ =>
-              return None
+        if (binOp.arg1 == stackPointer) {
+          val rhs: Expr = evaluateExpression(binOp.arg2, n, constantProp)
+          mmm.findStackObject(rhs.asInstanceOf[BitVecLiteral].value) match
+            case Some(obj: MemoryRegion) => Some(obj)
+            case _                       => None
         } else {
-            mmm.findObject(lhs.asInstanceOf[BitVecLiteral].value, "data") match
-                case Some(obj: MemoryRegion) =>
-                  return Some(obj)
-                case _ =>
-                return None
-            }
-        None
+          val evaluation: Expr = evaluateExpression(binOp, n, constantProp)
+          if (!evaluation.isInstanceOf[BitVecLiteral]) {
+            return None
+          }
+          mmm.findDataObject(evaluation.asInstanceOf[BitVecLiteral].value) match
+            case Some(obj: MemoryRegion) => Some(obj)
+            case _                       => None
```

Comment:
This is redundant? If findDataObject doesn't return a value of type Some[DataRegion] (and DataRegion is a subclass of MemoryRegion) it will return None so there isn't a need to match it?